src/Servers/Connections.Abstractions/src/IConnectionFactory.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Connections
+{
+    /// <summary>
+    /// A factory abstraction for creating connections to an endpoint.
+    /// </summary>
+    public interface IConnectionFactory
+    {
+        /// <summary>
+        /// Creates a new connection to an endpoint
+        /// </summary>
+        /// <param name="endPoint">The <see cref="EndPoint"/> to connect to.</param>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests. The default value is <see cref="CancellationToken.None" />.</param>
+        /// <returns>
+        /// A <see cref="ValueTask{TResult}"/> that represents the asynchronous connect.
+        /// The <see cref="Task{TResult}.Result"/> property returns a <see cref="ConnectionContext"/> for the new connection.
```

Comment:
yuck, let's not encourage use of `Task.Result`. Even if other docs do this, it's bad :). I would reword the whole thing in to:

```csharp
/// A <see cref="ValueTask{TResult}" /> that represents the asynchronous connect, yielding the <see cref="ConnectionContext" /> for the new connection when completed.
```

Sorry, can't make it a suggestion since it's a multi-line change.

Reply:
Clearly I wasn't careful enough when copying the old IConnectionFactory doc comments. I should have at least put *Value*Task{TResult}.Result in there! jk